contracts/mainnet/connectors/notional/main.sol

```diff
@@ -655,6 +655,139 @@ abstract contract NotionalResolver is Events, Helpers {
 		);
 	}
 
+	/// @notice Mints sNOTE from the underlying BPT token.
+	/// @dev Mints sNOTE from the underlying BPT token.
+	/// @param bptAmount is the amount of BPT to transfer from the msg.sender.
+	function mintSNoteFromBPT(uint256 bptAmount)
+		external
+		payable
+		returns (string memory _eventName, bytes memory _eventParam)
+	{
+		if (bptAmount == type(uint256).max)
+			bptAmount = bpt.balanceOf(address(this));
+
+		approve(bpt, address(staking), bptAmount);
+
+		staking.mintFromBPT(bptAmount);
+
+		_eventName = "LogMintSNoteFromBPT(address,uint256)";
+		_eventParam = abi.encode(address(this), bptAmount);
+	}
+
+	/// @notice Mints sNOTE from some amount of NOTE and ETH
+	/// @dev Mints sNOTE from some amount of NOTE and ETH
+	/// @param noteAmount amount of NOTE to transfer into the sNOTE contract
+	/// @param minBPT slippage parameter to prevent front running
+	function mintSNoteFromETH(
+		uint256 noteAmount,
+		uint256 ethAmount,
+		uint256 minBPT
+	)
+		external
+		payable
+		returns (string memory _eventName, bytes memory _eventParam)
+	{
+		if (noteAmount == type(uint256).max)
+			noteAmount = note.balanceOf(address(this));
+
+		if (ethAmount == type(uint256).max) ethAmount = address(this).balance;
+
+		approve(note, address(staking), noteAmount);
+
+		staking.mintFromETH{ value: ethAmount }(noteAmount, minBPT);
+
+		_eventName = "LogMintSNoteFromETH(address,uint256,uint256,uint256)";
+		_eventParam = abi.encode(address(this), ethAmount, noteAmount, minBPT);
+	}
+
+	/// @notice Mints sNOTE from some amount of NOTE and WETH
+	/// @dev Mints sNOTE from some amount of NOTE and WETH
+	/// @param noteAmount amount of NOTE to transfer into the sNOTE contract
+	/// @param wethAmount amount of WETH to transfer into the sNOTE contract
+	/// @param minBPT slippage parameter to prevent front running
+	function mintSNoteFromWETH(
+		uint256 noteAmount,
+		uint256 wethAmount,
+		uint256 minBPT
+	)
+		external
+		payable
+		returns (string memory _eventName, bytes memory _eventParam)
+	{
+		if (noteAmount == type(uint256).max)
+			noteAmount = note.balanceOf(address(this));
+
+		if (wethAmount == type(uint256).max)
+			wethAmount = weth.balanceOf(address(this));
+
+		approve(note, address(staking), noteAmount);
+		approve(weth, address(staking), wethAmount);
+
+		staking.mintFromWETH(noteAmount, wethAmount, minBPT);
+
+		_eventName = "LogMintSNoteFromWETH(address,uint256,uint256,uint256)";
+		_eventParam = abi.encode(address(this), noteAmount, wethAmount, minBPT);
+	}
+
+	/// @notice Begins a cool down period for the sender, this is required to redeem tokens
+	/// @dev Current cool down period is set to 1296000 seconds (15 days)
```

Comment:
Maybe dont leave this comment in here, we might change it in the future.

Reply:
removed